components/automate-ui/src/app/pages/project/list/project-list.component.ts

```diff
@@ -151,6 +148,16 @@ export class ProjectListComponent implements OnInit, OnDestroy {
     return '';
   }
 
+  getButtonText(): string {
+    if (this.applyRulesInProgress) {
+      return `Updating Projects ${Math.round(this.percentageComplete * 100)}%...`;
+    }
+    if (this.projectsHaveStagedChanges) {
+      return 'Update Projects';
+    }
+    return 'Projects Up-To-Date';
+  }
+
```

Comment:
side note -- as mentioned in one of @msorens' PRs recently -- I think it's preferrable to keep all text-y things in the template. Looking at the `component.html`, all we know is that the button is `getButtonText()`, we then have to lookup what this can be in the `component.ts`. here, however, we'd have to nest ternary operators in the `component.html`, and I suppose that's worse 😄